.github/workflows/front.yml

```diff
@@ -24,6 +24,11 @@ jobs:
 
       - name: Install dependencies
         run: yarn --cwd front install
+      
+      - name: Build project
```

Comment:
If the unit test fail there is no point to build. keep the build last in the chain

Reply:
Usually, it's best to build before the unit tests: https://softwareengineering.stackexchange.com/questions/403545/whats-the-proper-order-of-stages-in-a-ci-build

Reply:
It would only make sense if the build was necessary to tests the new changes. Here the test do not need the build at all. javascript is not compiled. The build is only here to optimize and minify.

What really matters is getting the pipeline the fail as early as possible, and getting feedback early and not waste time building something that’s not going to be used because the tests fail afterward.

> Optimise to quickly notify the developer of relevant feedback. (the total time between the developer pushing the code in and the developer actually becoming aware of the feedback).

> schedule them so that the tests most likely to fail, are run soonest.

Reply:
Actually the linter is run during the build, that is why I moved it here, because I think we should run the linter before the tests. What do you think ?

Reply:
Yeah, I guess the best order would be linter—> unit tests —> build

Maybe remove eslint from `vite.config.ts` and add it back as a job in the pipeline
@multun ?

Reply:
@anisometropie 
> Maybe remove eslint from vite.config.ts and add it back as a job in the pipeline

 there are multiple advantages to adding eslint to vite.config.ts:
 - eslint always runs during local builds using the same configuration as in the CI system
 - warnings and errors are reported in the dev server through vite-plugin-checker
 - vite runs eslint in parallel with the build. on my machine, adding eslint there does not increase build times.

> It would only make sense if the build was necessary to tests the new changes. Here the test do not need the build at all. javascript is not compiled. The build is only here to optimize and minify.

tsx has to be compiled to ts, ts has to be compiled to js, jsx has to be compiled to js, imports have to be resolved, and all of these things are done even for running tests

> Yeah, I guess the best order would be linter—> unit tests —> build

Does it really matter all that much?
 - if the build does not pass, unit tests usually cannot run, and the linter might get lost. that's why it usually runs first. Running the build first avoids chasing linter warnings when your app does not even build.
 - steps don't have to depend on the status on others. you can very much always run the linter, even if the build failed.